mat2vec/processing/pubmed_to_lines.py

```diff
@@ -0,0 +1,133 @@
+"""Converts PubMed abstracts to a file with one abstract per line."""
+import argparse
+import datetime
+import gzip
+import logging
+import multiprocessing
+import os
+import process
+import time
+import xml.etree.cElementTree as ET
+
+from functools import partial
+from typing import Tuple
+
+
+logging.basicConfig(format="%(asctime)s : %(levelname)s : %(message)s",
+                    level=logging.INFO)
+
+# Approximate total number of abstracts so we can estimate processing time
+# without having to read all files beforehand.
+TOTAL_DOCS = 32000000
+
+
+def abstracts_to_lines_fn(
+        args: Tuple[str],
+        text_processor: process.MaterialsTextProcessor) -> int:
+    """Write abstracts in a PubMed XML file to a line-separated text file.
+
+    Args:
+        args: A tuple containing the input and output file paths.
+        text_processor: an instance of process.MaterialsTextProcessor.
+
+    Returns:
+        doc_count: number of abstracts written to the file.
+    """
+    input_file, output_file = args
+    fout = open(output_file, 'w')
+    doc_count = 0
+    with gzip.open(input_file, 'rt') as f:
+        text = f.read()
+        tree = ET.ElementTree(ET.fromstring(text))
+        root = tree.getroot()
+
+        for abstract_text in root.iterfind('.//AbstractText'):
+            doc_text = abstract_text.text
+            if doc_text is None:
+                continue
+            doc_tokens, _ = text_processor.process(doc_text)
+            doc_text = ' '.join(doc_tokens)
+
+            fout.write('{}\n'.format(doc_text))
+            doc_count += 1
+            # if doc_count >= 1000:
+            #    break
+
+    fout.close()
+    return doc_count
+
+
+def abstracts_to_lines(input_folder: str, output_folder: str,
+                       n_processes: int) -> None:
+    """Write abstracts in PubMed XML files to line-separated text files.
+
+    Args:
+        input_folder: Path to the folder containing the PubMed XML files.
+        output_folder: Path to the folder to write the raw text files.
+        n_processes: Number of processes to read and write the files in
+            parallel.
+    """
+
+    start_time = time.time()
+    input_files = []
+    output_files = []
+    for file_name in os.listdir(input_folder):
```

Comment:
Since the files share the same basename, maybe extract the basename separately and then add the directories and extensions for clarity.
```suggestion
    files = [os.path.splitext(x)[0] for x in os.listdir(input_folder) if x.endswith('.xml.gz')]
    input_files = [os.path.join(input_folder, x + '.xml.gz') for x in files]
    output_files = [os.path.join(output_folder, x + '.txt') for x in files]
```